enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,225 @@
+linux Traffic Control (TC) tool `tc`. It relies on SR-IOV technology to control
+the SR-IOV Virtual Functions (VF) via VF representor port that is plugged
+into OpenvSwitch bridge. It improves the utilization of available bandwidth
+on underlying NICs by moving CPU intensive network packet processing from
+OVS software to hardware.
+
+## Motivation
+
+Enable high-performance datapath on primary pod interface with production-ready
+OVN-Kubernetes.
+
+### Goals
+
+Enable OpenFlow hardware offload for OVN datapath
+
+### Non-Goals
+
+Using hardware offload for any part of the OpenShift control plane
+
+## Proposal
+
+OVS Hardware Offload accelerates the OVS software flows with a SR-IOV NIC
+that works in `switchdev` mode. It requires configurations in both OpenvSwitch
+daemon and SR-IOV hardware NIC.
+
+- OpenvSwitch daemon is managed by Cluster Network Operator (CNO) and runs in
+`ovs-node` DaemonSet. A boolean configurable field will be added in CNO
+`Network` Custom Resource Definition `spec.defaultNetwork.ovnKubernetesConfig`
+to enable or disable OVS Hardware Offload feature for OpenvSwitch daemon.
+
+- SR-IOV hardware NIC is managed by SR-IOV Network Operator and will be
+provisioned with a new device type `switchdev`. The initial supported NIC
+model will be Mellanox MT27800 Family [ConnectX-5] 25GbE dual-port SFP28.
+
+Besides CNO and SR-IOV Operator changes, there are several other dependencies.
+
+- OVS Hardware Offload will only be supported on bare metal deployment.
+
+- Multus supports overwriting cluster network definition by specifying a
+customized default net-attach-def in pod annotation. Customized net-attach-def
+will still use `ovn-k8s-cni-overlay` CNI plugin, but with an special annotation
+that requests a SR-IOV device.
+
+- RHCOS image shall be based on RHEL-8.3 for fully support of OVS Hardware
+Offload with connection track feature.
+
+- OpenvSwitch package version 2.13.
+
+- To use OVS Hardware Offload with link redundency, a bond interface needs to
+be created and used as geneve endpoint of OVS bridge. Mellanox SR-IOV VF-LAG
+provides hardware link-aggregation (LAG) for hardware offloaded ports. It
+allows offloading of OVS rules to a linux `bond` network device that combines
+two idential and hardware offload capable Physical Functions (PFs). With the
+SR-IOV VF LAG capability, the NIC PFs can receive the rules that the OVS tries
+to offload to the linux bond net device and offload them to the hardware
+e-switch.
+
+#### OVN-Kubernetes
+
+- OVS Hardware Offload is a [feature available](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/ovs_offload.md) in OVN-Kubernetes upstream.
+
+- `ovn-k8s-cni-overlay` CNI plugin configures the pod interface by moving the VF
+to pod namespace and plugging VF representor to `br-int` OVS bridge. The
+relevant [patches](https://github.com/openshift/ovn-kubernetes/commit/6c96467d0d3e58cab05641293d1c1b75e5914795) are available downstream.
+
+#### Cluster Network Operator
+
```

Comment:
by the way what is doing the kubernetes configuration. We need topology manager with single-numa node to have the max performance.

Reply:
Topology Manager is not configured via cluster network operator, it's configured by applying `KubeletConfig` and `MachineConfig` manually or using `performance-addon-operator`.